### ISSUE TITLE
Stream does not start when video is black (RTMP only)

### ISSUE THREAD

**Issue by nangavar** [AirenSoft/OvenMediaEngine#195]
**Describe the bug**
When I try to start playing a WebRTC stream that comes from a RTMP provider, it won't start playing if the video is black in this moment.

**To Reproduce**
1. Stream trough RTMP a black video with OBS configured as docs.
2. Try to play this stream through WebRTC in OvenPlayer
3. You won't be able to play it until something appears on the video.

**Expected behavior**
OvenPlayer should start playing the stream, even if it is black video.

**Logs**
Nothing special on logs. Seems that it should work.

**Server (please complete the following information):**
 - OS: Ubuntu 18.04
 - OvenMediaEngine Version: 0.10.7
 - Branch: Master Brach

**Player (please complete the following information):**
 - Device: MAC, PC
 - OS: WIN10 - macOS - Android - iOS
 - Browser Chrome - Safari
 - Version 0.9

**Additional context**
- I have no problems playing a black video stream trough RTSP.
- If the stream was already playing, no problems, you will see the black video normally.

**Comment by getroot** [comment]
![image](https://user-images.githubusercontent.com/17779085/93729251-9cd9b600-fbfe-11ea-9c1f-73d8bc70e079.png)
![image](https://user-images.githubusercontent.com/17779085/93729253-a06d3d00-fbfe-11ea-98c2-2378c95a9ddc.png)
I can't reproduce this.
Does your OBS have a special function or is the function that does not transmit streams on a black screen is turned on? I can't understand this bug.

**Comment by nangavar** [comment]
Hi getroot!
Nothing special on my OBS configs to reproduce the bug. 
Are you transcoding video on OME server? I'm bypassing video on my OME config, maybe this is the difference. In the moment I add some content to the video then OvenPlayer starts to play.

**Comment by getroot** [comment]
Hmm...
I left only the bypass encoding setting in Server.xml and tried again, but it works fine too.

In fact, OvenMediaEngine doesn't know if a stream is input from RTMP whether it is a black screen or not. It is recognized only as an h.264 NAL unit. This is a scenario that can only happen if the encoder is not sending a stream when it is black screen.

When you click [Start Streaming] in OBS, is it outputted in the log of OME that the stream is created well? Are there any other problems with the log?

**Comment by getroot** [comment]
Oh, have you ever set "keyframe interval" to 1 in OBS? I think OBS can send keyframes very occasionally on black screen. Since WebRTC also starts playing after receiving the keyframe, it may seem that playback fails in OME's bypass setting.

**Comment by nangavar** [comment]
> When you click [Start Streaming] in OBS, is it outputted in the log of OME that the stream is created well? Are there any other problems with the log?

Yes, everything normal at OME side.

> Oh, have you ever set "keyframe interval" to 1 in OBS? I think OBS can send keyframes very occasionally on black screen. Since WebRTC also starts playing after receiving the keyframe, it may seem that playback fails in OME's bypass setting.

Yes, I had set up keyframe interval to 1 but it hadn't worked.
I agree that it seems to be a keyframe interval issue, the decoder on player side seems to be waiting for the keyframe in order to start playing.
I have tested streaming to your Korea's server with my OBS setup and it works, but I don't know if you're bypassing video on this OME. 

Are you sure you're testing with a complete black video? I tested with an AWS Elemental encoder and I got the same behaviour. I will continue making tests and will let you know if I find something.

Many thanks!

**Comment by getroot** [comment]
The settings below are the settings of the dev2 server.
Stream_o is playing at http://demo.ovenplayer.com/ome_demo.html. As you can see, stream_o stream bypasses video and audio encodes opus. And dev2 has a very new master branch installed (I haven't released it yet, but I am testing HEVC here).

`[root@dev2 conf]# cat Server.xml
<?xml version="1.0" encoding="UTF-8" ?>

<Server version="7">
        <Name>OvenMediaEngine</Name>
        
        <Type>origin</Type>
        
        <IP>*</IP>

        <Bind>
                <Providers>
                        <RTMP>
                                <Port>1935</Port>
                        </RTMP>
                        <MPEGTS>
                                <Port>4000-4005/udp</Port>
                        </MPEGTS>
                </Providers>

                <Publishers>
                        <OVT>
                                <Port>9000</Port>
                        </OVT>
                        <HLS>
                                <Port>80</Port>
                                <TLSPort>443</TLSPort>
                        </HLS>
                        <DASH>
                                <Port>80</Port>
                                <TLSPort>443</TLSPort>
                        </DASH>
                        <WebRTC>
                                <Signalling>
                                        <Port>3332</Port>
                                        <TLSPort>3333</TLSPort>
                                </Signalling>
                                <IceCandidates>
                                        <IceCandidate>*:10000-10005/udp</IceCandidate>
                                </IceCandidates>
                        </WebRTC>
                </Publishers>
        </Bind>

        <VirtualHosts>
                <VirtualHost include="VHost*.xml" />
                <VirtualHost>
                        <Name>default</Name>

                        <Domain>
                                <Names>
                                        <Name>*.airensoft.com</Name>
                                </Names>
                                <TLS>
                                        <CertPath>/etc/pki/tls/private/_airensoft_com.crt</CertPath>
                                        <ChainCertPath>/etc/pki/tls/certs/_airensoft_com_chain.crt</ChainCertPath>
                                        <KeyPath>/etc/pki/tls/private/_airensoft_com.key</KeyPath>
                                </TLS>
                        </Domain>

                        <Applications>
                                <Application>
                                        <Name>app</Name>
                                        <Type>live</Type>
                                        <Encodes>
                                                <Encode>
                                                        <Name>bypass</Name>
                                                        <Video>
                                                                <Bypass>true</Bypass>
                                                        </Video>
                                                </Encode>
                                                <Encode>
                                                        <Name>hevc</Name>
                                                        <Video>
                                                                <Codec>h265</Codec>
                                                                <Width>1280</Width>
                                                                <Height>720</Height>
                                                                <Bitrate>500000</Bitrate>
                                                                <Framerate>30</Framerate>
                                                        </Video>
                                                </Encode>
                                                <Encode>
                                                        <Name>opus</Name>
                                                        <Audio>
                                                                <Codec>opus</Codec>
                                                                <Bitrate>128000</Bitrate>
                                                                <Samplerate>48000</Samplerate>
                                                                <Channel>2</Channel>
                                                        </Audio>
                                                </Encode>

                                        </Encodes>
                                        <Streams>
                                                <Stream>
                                                        <Name>${OriginStreamName}_o</Name>
                                                        <Profiles>
                                                                <Profile>bypass</Profile>
                                                                <Profile>opus</Profile>
                                                        </Profiles>
                                                </Stream>
                                                <Stream>
                                                        <Name>${OriginStreamName}_hevc</Name>
                                                        <Profiles>
                                                                <Profile>hevc</Profile>
                                                                <Profile>opus</Profile>
                                                        </Profiles>
                                                </Stream>
                                        </Streams>
                                        <Providers>
                                                <RTMP />
                                                <MPEGTS>
                                                        <StreamMap>
                                                                <Stream>
                                                                        <Name>stream_${Port}</Name>
                                                                        <Port>4000-4005</Port>
                                                                </Stream>
                                                        </StreamMap>
                                                </MPEGTS>
                                        </Providers>
                                        <Publishers>
                                                <ThreadCount>8</ThreadCount>
                                                <HLS>
                                                        <SegmentDuration>5</SegmentDuration>
                                                        <SegmentCount>3</SegmentCount>
                                                        <CrossDomain>
                                                                <Url>*</Url>
                                                        </CrossDomain>
                                                </HLS>
                                                <DASH>
                                                        <SegmentDuration>5</SegmentDuration>
                                                        <SegmentCount>3</SegmentCount>
                                                        <CrossDomain>
                                                                <Url>*</Url>
                                                        </CrossDomain>
                                                </DASH>
                                                <LLDASH>
                                                        <SegmentDuration>5</SegmentDuration>
                                                        <CrossDomain>
                                                                <Url>*</Url>
                                                        </CrossDomain>
                                                </LLDASH>
                                                <WebRTC>
                                                        <Timeout>30000</Timeout>
                                                </WebRTC>
                                        </Publishers>
                                </Application>
                        </Applications>
                </VirtualHost>
        </VirtualHosts>
</Server>
`

**Comment by nangavar** [comment]
Hi getroot!
Finally I found the difference. 
Setting the H264 profile to "baseline" in OBS is the only option with that the player will start playing a black video. "Main" or "High" seems that won't send the keyframe if video is black. So I set the H264 profile to baseline and works.

I think we can close this issue!
And I'm sorry, I wasn't aware about this difference on my OBS settings. 
Many thanks for everything!!!
